searchlib/src/vespa/searchlib/attribute/attributeiterators.hpp

```diff
@@ -143,8 +191,18 @@ template <typename PL>
 std::unique_ptr<BitVector>
 FilterAttributePostingListIteratorT<PL>::get_hits(uint32_t begin_id) {
     BitVector::UP result(BitVector::create(begin_id, getEndId()));
-    for (; _iterator.valid() && _iterator.getKey() < getEndId(); ++_iterator) {
-        result->setBit(_iterator.getKey());
+    if constexpr (is_tree_iterator_v<PL>) {
+        auto end_itr = _iterator;
```

Comment:
This looks like the same code as on lines 142 - 148. Consider refactor into common function.